core/store/src/trie/mem/parallel_loader.rs

```diff
@@ -213,7 +214,7 @@ impl ParallelMemTrieLoader {
         arena: &mut impl Arena,
     ) -> Result<MemTrieNodeId, StorageError> {
         // Figure out which range corresponds to the prefix of this subtree.
-        let (start, end) = subtree_to_load.to_iter_range(self.shard_uid);
+        let (start, end) = subtree_to_load.to_iter_range(self.shard_uid_db_prefix.0);
 
         // Load all the keys in this range from the FlatState column.
```

Comment:
Not sure about that. Maybe it would read entire parent shard while we only want child shard.
And it reads FlatState so we might want to synchronize on changes here.

Reply:
Marked as `TODO(reshardingV3)`, to be handled in a separate PR.